clusters/resource_cluster.go

```diff
@@ -57,71 +58,119 @@ func ZoneDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 }
 
-func resourceClusterSchema() map[string]*schema.Schema {
-	return common.StructToSchema(Cluster{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
-		s["spark_conf"].DiffSuppressFunc = SparkConfDiffSuppressFunc
-		common.MustSchemaPath(s, "aws_attributes", "zone_id").DiffSuppressFunc = ZoneDiffSuppress
-		common.MustSchemaPath(s, "gcp_attributes", "use_preemptible_executors").Deprecated = "Please use 'availability' instead."
+type ClusterResourceProvider struct{}
 
-		common.MustSchemaPath(s, "init_scripts", "dbfs").Deprecated = DbfsDeprecationWarning
+func (ClusterResourceProvider) UnderlyingType() compute.ClusterSpec {
+	return compute.ClusterSpec{}
+}
 
-		// adds `library` configuration block
-		s["library"] = common.StructToSchema(libraries.ClusterLibraryList{},
-			func(ss map[string]*schema.Schema) map[string]*schema.Schema {
-				ss["library"].Set = func(i any) int {
-					lib := libraries.NewLibraryFromInstanceState(i)
-					return schema.HashString(lib.String())
-				}
-				return ss
-			})["library"]
+func (ClusterResourceProvider) Aliases() map[string]string {
+	return map[string]string{"cluster_mount_infos": "cluster_mount_info"}
+}
 
-		s["autotermination_minutes"].Default = 60
-		s["cluster_id"] = &schema.Schema{
+func (ClusterResourceProvider) CustomizeSchema(s map[string]*schema.Schema) map[string]*schema.Schema {
+	common.CustomizeSchemaPath(s, "enable_elastic_disk").SetComputed()
+	common.CustomizeSchemaPath(s, "enable_local_disk_encryption").SetComputed()
+	common.CustomizeSchemaPath(s, "node_type_id").SetComputed().SetConflictsWith([]string{"driver_instance_pool_id", "instance_pool_id"})
+	common.CustomizeSchemaPath(s, "driver_node_type_id").SetComputed().SetConflictsWith([]string{"driver_instance_pool_id", "instance_pool_id"})
+	common.CustomizeSchemaPath(s, "driver_instance_pool_id").SetComputed().SetConflictsWith([]string{"driver_node_type_id", "node_type_id"})
+	common.CustomizeSchemaPath(s, "ssh_public_keys").SetMaxItems(10)
+	common.CustomizeSchemaPath(s, "init_scripts").SetMaxItems(10).AddNewField("abfss", common.StructToSchema(InitScriptStorageInfo{}, nil)["abfss"]).AddNewField("gcs", common.StructToSchema(InitScriptStorageInfo{}, nil)["gcs"])
+	common.CustomizeSchemaPath(s, "init_scripts", "dbfs").SetDeprecated(DbfsDeprecationWarning)
+	common.CustomizeSchemaPath(s, "init_scripts", "dbfs", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "s3", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "volumes", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "workspace", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "workload_type", "clients").SetRequired()
+	common.CustomizeSchemaPath(s, "workload_type", "clients", "notebooks").SetDefault(true)
+	common.CustomizeSchemaPath(s, "workload_type", "clients", "jobs").SetDefault(true)
+	common.CustomizeSchemaPath(s).AddNewField("idempotency_token", &schema.Schema{
```

Comment:
If it's not part of the `ClusterSpec` and isn't returned back by it, then we may get a permanent drift? And because iit's `ForceNew`, then it will try to recreate a cluster.

Reply:
This won't be impacted by this PR, but it will be when @tanmay-db migrates the resource implementation to the Go SDK.